docs/anti-patterns.mdx

```diff
@@ -4,35 +4,110 @@ title: Cadence Anti-Patterns
+
+## Auth References should be avoided
+
+### Problem
+
+[`auth` references](/language/references) should be avoided unless necessary:
+it allows downcasting any restricted type to its unrestricted type.
+The parent type could expose functionality that was not intended to be exposed.
+If the auth keyword is used on local variables they will be references.
+References are ephemeral and cannot be stored.
+This prevents any reference casting to be stored under account storage.
+However, if the auth keyword is used to store a public capability serious harm
+could be achieved since the value could be downcasted to a type that has functionality and values altered.
 
-If those events are meant to indicate actions taken using a single instance of that resource (for example an admin object, switchboard, or registry), or instances created through a particular workflow, then this will make event log search and management harder because events from other instances may be mixed in with the ones that you wish to examine.
+### Example
 
-To fix this, if there should be only a single instance of the resource it should be created and `link()`ed to a public path in an admin account's storage during the contracts's initializer.
+A common pattern in many NFT smart contracts is to include a public borrow function
+that borrows an auth reference to an NFT ([See Top Shot for an example](https://github.com/dapperlabs/nba-smart-contracts/blob/master/contracts/TopShot.cdc#L889-L906)).
+This allows anyone to access the stored metadata or extra fields that weren't part of the NFT standard.
+While this is fine in most scenarios, some NFTs might have privledged functions that shouldn't
+be exposed this way, so be careful when copying NFT projects that use this pattern. 
 
-### Named Value Fields Are Preferable
+## Events From Resources Might Not Be Unique
 
-Where contracts, resources, and scripts all have to repeatedly refer to the same constant values, entering these manually is a potential source of error. These are known as [magic values](<https://en.wikipedia.org/wiki/Magic_number_(programming)>).
+### Problem
 
-Rather than typing the values manually in functions, transactions and scripts, store the values once in fields of the contract responsible and then access them via those fields. This is the [Named Value Field pattern](design-patterns/#named-value-field)
+A public function on a contract that creates a resource can be called by any account.
+If that resource has functions that emit events, any account can create an instance of that resource and emit those events.
+If those events are meant to indicate actions taken using a single instance of that resource
+(for example an admin object, switchboard, or registry),
+or instances created through a particular workflow, then this will make event log search
+and management harder because events from other instances may be mixed in with the ones that you wish to examine.
```

Comment:
Public functions in a contract can be called by any account. That being said, if that function creates a resource, and that resource has functions that emit events, that means any account can create an instance of that resource and emit those events. If those events are meant to indicate actions taken using a single instance of that resource (eg. admin object, registry), or instances created through a particular workflow, it's possible that events from other instances may be mixed in with the ones you're querying for - making the event log search and management more cumbersome.